src/main/java/org/apache/cassandra/distributed/shared/Versions.java

```diff
@@ -101,11 +102,14 @@ public Version get(String full)
 
     public Version get(Semver version)
     {
-        return versions.get(first(version))
-                       .stream()
+        Supplier<RuntimeException> onError = () -> new RuntimeException("No version " + version.getOriginalValue() + " found");
+        List<Version> versions = this.versions.get(first(version));
+        if (versions == null)
+            throw onError.get();
+        return versions.stream()
                        .filter(v -> version.equals(v.version))
                        .findFirst()
-                       .orElseThrow(() -> new RuntimeException("No version " + version.getOriginalValue() + " found"));
+                       .orElseThrow(onError);
```

Comment:
This was found in another ticket, if the version doesn't have the match, we return null and throw a NPE, so handle that the same as `orElseThrow`